src/Maestro/Maestro.Web/Startup.cs

```diff
@@ -392,6 +403,52 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env)
                 app.UseHttpsRedirection();
             }
 
+            // Add security headers
+            app.Use(
+                (ctx, next) =>
+                {
+                    ctx.Response.OnStarting(() =>
+                    {
+                        if (!ctx.Response.Headers.ContainsKey("X-XSS-Protection"))
+                        {
+                            ctx.Response.Headers.Add("X-XSS-Protection", "1");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("X-Frame-Options"))
+                        {
+                            ctx.Response.Headers.Add("X-Frame-Options", "DENY");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("X-Content-Type-Options"))
+                        {
+                            ctx.Response.Headers.Add("X-Content-Type-Options", "nosniff");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("Referrer-Policy"))
+                        {
+                            ctx.Response.Headers.Add("Referrer-Policy", "no-referrer-when-downgrade");
+                        }
+
+                        if (!ctx.Response.Headers.ContainsKey("Cache-Control"))
+                        {
+                            ctx.Response.Headers.Add("Cache-Control", "no-cache, no-store, must-revalidate, proxy-revalidate");
```

Comment:
Wait... why are we removing all ability for caching here? That doesn't seem like a security issue, and the JS files are pretty big, so we really would prefer people cache those.

Reply:
The two big security risks are in replay attacks from cache and data retrieval from cache (both on a compromised client machine). We don't have any data that could be usefully retrieved (e.g. credit card information), so we would be ok skipping "no-store". That should remove the potential performance impact since "no-cache" is a misnomer, all it does is require a check before using a cached copy.

(See the "Browser Cache" section on here for more details on the potential attacks: https://owasp.org/www-community/OWASP_Application_Security_FAQ)

Reply:
We don't have anything sensitive in any of the pages, I don't think we need to be globally messing with the caching headers.  The justifications in that document seem to only apply to "sensitive content", which we don't really have, especially since the cookie is always going to be stored (that's sort of it's purpose in life) and it can be used to get any of the information again.

If we want to mess around with caching for functionality reasons (if we are over or under caching some piece of information, like the CSS), then that's totally fine, but it should be a different PR, since it doesn't have anything to do with security.  And it should be on a per-item basis (API's probably don't want to cache, the static content html, jss, and css we probably DO want to cache... but that means no global headers can really accomplish that)

Reply:
That's all reasonable. I wanted to err on the side of more secure whenever possible, but I agree that we don't have anything sensitive enough to require changes to the caching.